File: Examples/VisualBasic/AsposePDF/Images/ExtractImages.vb
Imports System.IO
Imports System
Imports Microsoft.VisualBasic
Imports Aspose.Pdf
Imports System.Drawing.Imaging
Namespace AsposePDF.Images
    Public Class ExtractImages
        Public Shared Sub Run()
            ' ExStart:ExtractImages
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdf_Images()

            ' Open document
            Dim pdfDocument As New Document(dataDir & Convert.ToString("ExtractImages.pdf"))

            ' Extract a particular image
            Dim xImage As XImage = pdfDocument.Pages(1).Resources.Images(1)

            Dim outputImage As New FileStream(dataDir & Convert.ToString("output.jpg"), FileMode.Create)

            ' Save output image
            xImage.Save(outputImage, ImageFormat.Jpeg)
            outputImage.Close()

            dataDir = dataDir & Convert.ToString("ExtractImages_out_.pdf")

            ' Save updated PDF file
            pdfDocument.Save(dataDir)
            ' ExEnd:ExtractImages
            Console.WriteLine(Convert.ToString(vbLf & "Images extracted successfully." & vbLf & "File saved at ") & dataDir)

        End Sub
    End Class
End Namespace

File: Examples/CSharp/AsposePDF/Images/ExtractImages.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using System.Drawing.Imaging;

namespace CSharp.AsposePdf.Images
{
    public class ExtractImages
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdf_Images();

            //open document
            Document pdfDocument = new Document(dataDir+ "ExtractImages.pdf");

            //extract a particular image
            XImage xImage = pdfDocument.Pages[1].Resources.Images[1];

            FileStream outputImage = new FileStream(dataDir + "output.jpg", FileMode.Create);

            //save output image
            xImage.Save(outputImage, ImageFormat.Jpeg);
            outputImage.Close();

            //save updated PDF file
            pdfDocument.Save(dataDir + "ExtractImages_out.pdf");
            
        }
    }
}